index.js

```diff
@@ -15,12 +14,12 @@ Sap.prototype.execute = function (args, callback) {
   var that = this;
   var requestParams = formatParams(args.params);
 
-  that.tokenPromise
-    .then(function (accessToken) {
+  that.authService.tokenPromise
+    .then(function (tokenData) {
       var options = {
         method: args.method,
         url: that.httpUri + '/' + that.version + '/' + args.path
-          + '?' + requestParams + 'access_token=' + accessToken,
+          + '?' + requestParams + 'access_token=' + tokenData.access_token,
```

Comment:
tokenPromise now resolves will full token object from sap (so we can use expiry_time as well as access_token)